bpf/init.sh

```diff
@@ -428,11 +439,17 @@ fi
 
 if [ "${TUNNEL_MODE}" != "<nil>" ]; then
 	ENCAP_DEV="cilium_${TUNNEL_MODE}"
-	ip link show $ENCAP_DEV || {
-		ip link add name $ENCAP_DEV address $(rnd_mac_addr) type $TUNNEL_MODE external || encap_fail
-	}
+
+	ip link show $ENCAP_DEV || create_encap_dev
 	ip link set $ENCAP_DEV mtu $MTU || encap_fail
 
+	if [ "${TUNNEL_PORT}" != "<nil>" ]; then
+		ip link show $ENCAP_DEV | grep "dstport $TUNNEL_PORT" || {
```

Comment:
Looks like this doesn't actually work, `ip link show` doesn't seem to display what `dstport` is being used, or any other driver-specific flags... I am actually not sure how is one supposed to check this. I would assume there is ought to be way to do it using netlink directly?

cc @borkmann @jrfastab

Reply:
`ip -details link show` 🚀